src/components/site-loading-indicator.tsx

```diff
@@ -1,37 +1,25 @@
 import { useI18n } from '@wordpress/react-i18n';
-import { useEffect } from 'react';
-import { useProgressTimer } from '../hooks/use-progress-timer';
-import ProgressBar from './progress-bar';
+import { useImportExport } from '../hooks/use-import-export';
+import ProgressBar, { ProgressBarWithAutoIncrement } from './progress-bar';
 
-export function SiteLoadingIndicator( { selectedSiteName }: { selectedSiteName?: string } ) {
+export function SiteLoadingIndicator( { selectedSite }: { selectedSite: SiteDetails } ) {
 	const { __ } = useI18n();
+	const { importState } = useImportExport();
+	const { [ selectedSite.id ]: currentProgress } = importState;
+	const isImporting = !! currentProgress?.progress;
 
-	const { progress, setProgress } = useProgressTimer( {
-		initialProgress: 10,
-		interval: 1500,
-		maxValue: 95,
-	} );
-
-	useEffect( () => {
-		const updateProgress = () => {
-			setProgress( ( prev ) => {
-				const increment = Math.random() * 10 + 5;
-				return Math.min( prev + increment, 95 );
-			} );
-		};
-
-		setProgress( 20 );
-		const interval = setInterval( updateProgress, 1000 );
-
-		return () => clearInterval( interval );
-	}, [ setProgress ] );
+	const statusMessage = isImporting ? currentProgress.statusMessage : __( 'Creating site...' );
 
 	return (
 		<div className="flex flex-col w-full h-full app-no-drag-region pt-8 overflow-y-auto justify-center items-center">
 			<div className="w-[300px] text-center">
-				<div className="text-black a8c-subtitle-small mb-4">{ selectedSiteName }</div>
-				<ProgressBar value={ progress } maxValue={ 100 } />
-				<div className="text-a8c-gray-70 a8c-body mt-4">{ __( 'Creating site...' ) }</div>
+				<div className="text-black a8c-subtitle-small mb-4">{ selectedSite.name }</div>
+				{ isImporting ? (
+					<ProgressBar value={ currentProgress.progress } maxValue={ 100 } />
+				) : (
+					<ProgressBarWithAutoIncrement initialValue={ 10 } increment={ 10 } maxValue={ 100 } />
```

Comment:
In the future, it would be nice to report the progress of site creation similar to how we do with import/export processes.